lib/auth/server_monitor.go

```diff
@@ -0,0 +1,161 @@
+// information about the system clock differences.
+func (a *Server) checkInventorySystemClocks(ctx context.Context) {
+	var counter int
+	var messages []string
+	a.inventory.Iter(func(handle inventory.UpstreamHandle) {
+		counter++
+		if counter >= systemClockMessagesLimit {
+			return
+		}
+
+		hello := handle.Hello()
+		handle.VisitInstanceState(func(ref inventory.InstanceStateRef) (update inventory.InstanceStateUpdate) {
+			if ref.LastHeartbeat != nil && ref.LastHeartbeat.GetLastMeasurement() != nil {
+				m := ref.LastHeartbeat.GetLastMeasurement()
+				diff := m.ControllerSystemClock.Sub(m.SystemClock) - m.RequestDuration/2
+				if diff > systemClockThreshold || -diff > systemClockThreshold {
+					slog.WarnContext(ctx, "server time difference detected",
+						"server", hello.GetServerID(),
+						"services", hello.GetServices(),
+						"difference", durationText(diff),
+					)
+					messages = append(messages, fmt.Sprintf(
+						" - %s[%s] is %s",
+						hello.GetServerID(),
+						types.SystemRoles(hello.GetServices()).String(),
+						durationText(diff),
+					))
+				}
+			}
+			return
+		})
+	})
+
+	if len(messages) > 0 {
+		err := upsertClockWarningGlobalNotification(ctx, a.Services, generateClockWarningNotificationMessage(messages, counter))
+		if err != nil {
+			slog.ErrorContext(ctx, "can't set notification about system clock issue", "error", err)
+		}
+	}
+}
+
+// upsertClockWarningGlobalNotification sets predefined global notification for notifying the issues with the cluster
+// servers related to the system clock difference in nodes.
+func upsertClockWarningGlobalNotification(ctx context.Context, notification services.Notifications, text string) error {
+	_, err := notification.UpsertGlobalNotification(ctx, &notificationsv1.GlobalNotification{
+		Kind:     types.KindGlobalNotification,
+		Version:  types.V1,
+		Metadata: &headerv1.Metadata{Name: systemClockNotificationWarningName},
+		Spec: &notificationsv1.GlobalNotificationSpec{
+			Matcher: &notificationsv1.GlobalNotificationSpec_All{
+				All: true,
+			},
+			Notification: &notificationsv1.Notification{
+				SubKind: types.NotificationDefaultWarningSubKind,
+				Spec:    &notificationsv1.NotificationSpec{},
+				Metadata: &headerv1.Metadata{
+					Name: systemClockNotificationWarningName,
+					Labels: map[string]string{
+						types.NotificationTitleLabel: text,
+					},
+				},
+			},
+		},
+	})
+	return trace.Wrap(err)
+}
+
+// generateClockWarningNotificationMessage formats the notification message with the inventory list.
+func generateClockWarningNotificationMessage(messages []string, total int) string {
+	message := "Incorrect system clock detected in the cluster, which may lead to certificate validation issues.\n" +
+		"Ensure that the clock is accurate on all nodes to avoid potential access problems.\n" +
+		"All comparisons are made with the Auth service system clock." +
+		"List of servers with a time drift: \n" + strings.Join(messages, "\n")
```

Comment:
```bash
root@3e255c82fc04:/go/teleport# ./build/tctl notifications ls --all
ID                                   Created             Expires             Title                                                                                                                                                                                                                                                                                                                 Labels
------------------------------------ ------------------- ------------------- --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ------
cluster-monitor-system-clock-warning 27 Nov 24 23:36 UTC 10 Dec 24 11:36 UTC Incorrect system clock detected in the cluster, which may lead to certificate validation issues.
Ensure that the clock is accurate on all nodes to avoid potential access problems.
All comparisons are made with the Auth service system clock.List of servers with a time drift:
 - test-id[Node] is 10h0m0s ahead
```

<img width="487" alt="image" src="https://github.com/user-attachments/assets/710808f8-8b5f-4657-860e-c516e525012e">

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/0da5b7cd-5820-4d8d-ae56-cb06139bcb4f">

looks like for the web, for spacing we should use something different

Reply:
I made some adjustments, moved message to `ContentLabel` and changed format to be readable inline

```bash
root@3e255c82fc04:/go/teleport# ./build/tctl notifications ls --all
ID                                   Created             Expires             Title                                          Labels
------------------------------------ ------------------- ------------------- ---------------------------------------------- ------
cluster-monitor-system-clock-warning 28 Nov 24 01:28 UTC 28 Dec 24 01:28 UTC Incorrect system clock detected in the cluster
```

<img width="488" alt="image" src="https://github.com/user-attachments/assets/2d67d08a-caf4-4bc7-96ac-0e48fb0f6968">
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/4628d485-e4ea-49bb-a56a-b94969ee6975">

Reply:
notifications as far as I understand aren't shown with `tsh status` and `tsh login` only cluster alerts, so you need to request them manually `tctl notifications ls --all`